lib/webidl2.js

```diff
@@ -97,7 +137,8 @@
       if (!tokens.length || tokens[0].type !== type) return;
       if (typeof value === "undefined" || tokens[0].value === value) {
         last_token = tokens.shift();
-        if (type === ID) last_token.value = last_token.value.replace(/^_/, "");
+        if (type === ID && last_token.value.charAt(0) === '_')
```

Comment:
I would prefer `last_token.value.startsWith("_")`, IMO it shouldn't be slower than `charAt`...